sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/sns/SnsIO.java

```diff
@@ -62,310 +58,272 @@
-     * more sophisticated credential protocol, then you should look at {@link
+     * Specify credential details and region to be used to write to SNS. If you need more
+     * sophisticated credential protocol, then you should look at {@link
      * Write#withSnsClientProvider(SnsClientProvider)}.
      */
-    public Write<T> withSnsClientProvider(
-        AwsCredentialsProvider credentialsProvider, String region) {
+    Write<T> withSnsClientProvider(AwsCredentialsProvider credentialsProvider, String region) {
+      checkNotNull(credentialsProvider, "credentialsProvider cannot be null");
+      checkNotNull(region, "region cannot be null");
       return withSnsClientProvider(credentialsProvider, region, null);
     }
 
     /**
-     * Specify {@link AwsCredentialsProvider} and region to be used to write to SNS. If you need
-     * more sophisticated credential protocol, then you should look at {@link
+     * Specify credential details and region to be used to write to SNS. If you need more
+     * sophisticated credential protocol, then you should look at {@link
      * Write#withSnsClientProvider(SnsClientProvider)}.
      *
-     * <p>The {@code serviceEndpoint} sets an alternative service host. This is useful to execute
-     * the tests with Kinesis service emulator.
+     * <p>The {@code serviceEndpoint} sets an alternative service host.
      */
-    public Write<T> withSnsClientProvider(
+    Write<T> withSnsClientProvider(
         AwsCredentialsProvider credentialsProvider, String region, URI serviceEndpoint) {
+      checkNotNull(credentialsProvider, "credentialsProvider cannot be null");
+      checkNotNull(region, "region cannot be null");
       return withSnsClientProvider(
           new BasicSnsClientProvider(credentialsProvider, region, serviceEndpoint));
     }
 
-    /**
-     * Provides configuration to retry a failed request to publish a message to SNS. Users should
-     * consider that retrying might compound the underlying problem which caused the initial
-     * failure. Users should also be aware that once retrying is exhausted the error is surfaced to
-     * the runner which <em>may</em> then opt to retry the current partition in entirety or abort if
-     * the max number of retries of the runner is completed. Retrying uses an exponential backoff
-     * algorithm, with minimum backoff of 5 seconds and then surfacing the error once the maximum
-     * number of retries or maximum configuration duration is exceeded.
-     *
-     * <p>Example use:
-     *
-     * <pre>{@code
-     * SnsIO.write()
-     *   .withRetryConfiguration(SnsIO.RetryConfiguration.create(5, Duration.standardMinutes(1))
-     *   ...
-     * }</pre>
-     *
-     * @param retryConfiguration the rules which govern the retry behavior
-     * @return the {@link Write} with retrying configured
-     */
-    public Write<T> withRetryConfiguration(RetryConfiguration retryConfiguration) {
```

Comment:
Please, deprecate this method and warn user explicitly that it should not be used anymore.